source/triggers/groups/triggerGroup.js

```diff
@@ -60,8 +60,9 @@ this.checkTriggers$ = function() {
     //  impossible - especially with a low frame rate.  In order to address 
     //  this, do another check after a future(0).
     if ( haveTriggerToFire ) {
-        this.future( 0 ).checkTriggersCallback$();
```

Comment:
The future(0) is there for a reason - and potentially important to avoid odd race conditions.  It basically says "wait until everything else has had a chance to go, and then check again."  

For example, imagine that the order of events in a particular frame is:

- Blockly program marks itself as done
- The trigger group checks its triggers
- The collision system sends an event telling us the rover has collided with an obstacle

When the trigger group runs, the collision system has not yet registered the collision, so the scenario would fail because the program ended (or might even succeed) when it should fail because of the collision.